tests/core/unit/registerCustomElement.ts

```diff
@@ -349,4 +372,35 @@ describe('registerCustomElement', () => {
 		const { display } = global.getComputedStyle(element);
 		assert.equal(display, 'block');
 	});
+
+	it('handles children being appended as document is still loading', async () => {
+		register(DelayedChildrenWidget);
+
+		Object.defineProperty(document, 'readyState', {
+			configurable: true,
+			get() {
+				return 'loading';
+			}
+		});
+		element = document.createElement('delayed-children-element');
+		document.body.appendChild(element);
+
+		let child = document.createTextNode('foo');
+		element!.appendChild(child);
+
+		Object.defineProperty(document, 'readyState', {
+			configurable: true,
+			get() {
+				return 'complete';
+			}
+		});
+		child = document.createTextNode('bar');
+		element!.appendChild(child);
+
+		waitFor(
```

Comment:
curious, do you need to return this to ensure that the test waits for the promise to be resolved?

Reply:
Yes 🤦‍♂ . This is actually broken in IE. Looking into it

Reply:
@agubler Ok it was still the test that was broken I just didn't fix it properly. Should be good now (pending CI)